src/util/ringdelaybuffer.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include "util/samplebuffer.h"
+#include "util/types.h"
+
+// TODO(davidchocholaty) Done documentation.
+
+/// The ring delay buffer allows moving with the reading position subject
+/// to certain rules. Another difference between the classic ring buffer is,
+/// that the ring delay buffer offers to read zero values
+/// which were not written by using the write method and write position.
+/// Both of these two specific properties are based on the cross-fading
+/// between changes of two delays.
+class RingDelayBuffer final {
+  public:
+    RingDelayBuffer(SINT bufferSize);
+
+    virtual ~RingDelayBuffer(){};
+
+    bool isFull() {
+        return getWriteAvailable() == 0;
+    }
+
+    bool isEmpty() const {
+        return m_readPos == m_writePos;
+    }
+
+    void clear() {
+        m_readPos = 0;
+        m_writePos = 0;
+
+        memset(m_buffer.data(), 0, sizeof(m_buffer));
```

Comment:
I'm not sure this is correct. I think `sizeof(m_buffer)` only yields the size of the control block (so 16 bytes on x64 because the control block is just a pointer and size).